test/distributed/test_c10d.py

```diff
@@ -2441,6 +2442,27 @@ def _test_nccl_backend(
             process_group, devices, device_ids, multi_device, gradient_as_bucket_view
         )
 
+    @requires_nccl()
+    @skip_if_not_multigpu
+    def test_nccl_backend_multi_device_ids_not_allowed(self):
+        int_devices = list(range(torch.cuda.device_count()))
+        devices = [torch.device("cuda:" + str(i)) for i in int_devices]
+        with self.assertRaises(AssertionError):
```

Comment:
I think its probably better to raise a `ValueError` when a user passes invalid arguments to an API.